### PR TITLE
Add offline first approach

### PR DESCRIPTION
**Pull request by denar90**
This issue is related to #1454 

@paulmillr if we agree on this API I can write some doc for offline first approach.

### CODE REVIEW

**Comment by Florian-R** [comment]
@denar90 Out of curiosity, why did you chose to add another option? Won't this be a better fit for a plugin?

File: lib/fs_utils/write.js [hunk 1]
```diff
@@ -162,6 +171,8 @@ const write = (fileList, config, joinConfig, optimizers, startTime) => {
   })).then(() => {
     return writeStatic(fileList, config, startTime);
   }).then(() => {
+    if (config.offlineFirst.enabled) return writeSWPreCache(config);
```
Comment (shvaikalesh) [inline lib/fs_utils/write.js]: Is there a need to wait for `writeStatic` to complete before calling `writeSWPreCache`? Can we do it in parallel?

**Comment by denar90** [comment]
@Florian-R It can be as separate plugin, but the reason why I propose to use it as part of brunch is offline approach for building applications. As I see the future, app will have to support offline approach be default. So having this behavior in core of brunch will be huge + for it. But I can be wrong and to hurry about using this approach so early in core. If you guys suggest just do it in scope of plugin it's fine by me. Just want to do what is better 😄

**Comment by shvaikalesh** [comment]
Brunch is opinionated build system, but having `GoogleChrome/sw-precache` built right in is too much. And it adds couple dependencies to Brunch itself and prolongs `npm i`. Separate plugin would be awesome.

@denar90, thank you for contribution.

**Pull request closed without merging**